### PR TITLE
Metrics + Prometheus + Grafana howto

### PR DESCRIPTION
**Pull request by tsegismont**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  implementation("io.vertx:vertx-micrometer-metrics:${vertxVersion}")
+  implementation("io.micrometer:micrometer-registry-prometheus:1.5.2")
+}
+
+application {
+  mainClassName = "io.vertx.howtos.metrics.MainVerticle"
+}
+
+tasks.wrapper {
+  gradleVersion = "5.2"
```
Comment (jponge) [inline build.gradle.kts]: Things move fast in Gradle land 😆 

You should try 6.8 (and `gradlew wrapper` to bump)

File: src/main/java/io/vertx/howtos/metrics/MainVerticle.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      });
+    });
+    // end::consumer[]
+
+    // tag::router[]
+    Router router = Router.router(vertx);
+    // end::router[]
+    // tag::handler[]
+    router.get("/greeting").handler(rc -> {
+      vertx.eventBus().<String>request("greetings", null, ar -> {
```
Comment (jponge) [inline src/main/java/io/vertx/howtos/metrics/MainVerticle.java]: How about using `Future` here? (nothing wrong at all with the callback approach)

File: README.adoc [hunk 3]
```diff
@@ -1,180 +1,1643 @@
...
+- generates greetings upon request on the `EventBus`
+- replies to `GET` requests on the `/greeting` path,
+- exposes metrics for Prometheus on the `/metrics` path.
 
-Projects should preferably have both Maven and Gradle build descriptors.
-An author who doesn't have knowledge of either of the tools can always ask for contributions.
+Then you will configure Prometheus to scrape the metrics and Grafana to visualize them.
 
-== Structure of a how-to
+The application consists in a single file, the `MainVerticle` class
```
Comment (jponge) [inline README.adoc]:
```suggestion
The application fits in just a single `MainVerticle` class
```

**Review by jponge — COMMENTED** [review]
Minor changes/comments, other than that it's looking great and useful!

**Comment by tsegismont** [comment]
@jponge done

**Comment by jponge** [comment]
Fantastic Thomas

**Pull request merged**